Feature/report tip commands

sellmark opened the pull request.

PogSmok committed:
Create EN.json

PogSmok committed:
Create PL.json

PogSmok committed:
Change hardcoded answers for JSON imports

PL.json - polish translations
EN.json - english defaults
Both files must be in the same folder as discord_bot.py, otherwise their paths must be changed (line 35 and 39)

PogSmok committed:
Format fix

PogSmok committed:
Create __init__.py

PogSmok committed:
Create PL.json

PogSmok committed:
Create EN.json

PogSmok committed:
Create translations.py

PogSmok committed:
Changes according to comments on pr

PogSmok committed:
Update EN.json

PogSmok committed:
Update PL.json

PogSmok committed:
Update EN.json

PogSmok committed:
Update PL.json

PogSmok committed:
Delete app/balancer/management/commands/EN.json

PogSmok committed:
Delete app/balancer/management/commands/PL.json

PogSmok committed:
Changed directory to app/balancer/

PogSmok committed:
Fixed queue_str error

sellmark committed:
Report Tip init with migrations. Self change dota id command

sellmark committed:
Unlock commands to channel

sellmark committed:
Further upgrades admin, website and some other commands

sellmark committed:
Report and tip understand mentions

sellmark committed:
Final updates

sellmark committed:
!adminhelp

sellmark committed:
!tips !reports

sellmark committed:
Changed voice chan name

sellmark committed:
Changed voice chan name

PogSmok committed:
Translation update

PogSmok committed:
Translation Update

PogSmok committed:
Update discord_bot.py

Komendy usunięte:
- afk ping
- join
- leave
- join
- list

Komendy zmienione:
- registration help
- who is 
- vote kick
- recent 
- top/bot
- roles
- help
- adminhelp

TODO:
Skończyć format stringa "game_start"

sellmark committed:
Merged translations, left some comments

sellmark committed:
Merged translations, left some comments

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -392,15 +386,14 @@ async def register_new_player(self, msg, name, mmr, dota_id):
-        )
+        await msg.channel.send(TRANSLATIONS[LANG]["welcome"].format(name, queue_channel))
 
+        ##TODO: finish translations for second msg
```
tutaj

PogSmok replied:
Nie rozmawialiśmy wcześniej, że wiadomość wysyłana na kanał, że ktoś się zapisał, jest zbędna? Potwierdzenie ma już w prywatnej wiadomości.

sellmark replied:
może z SATO rozmawiałeś, ja nie kojarzę, można wywalić, nie wiedziałem że jest za tym decyzja

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -453,29 +444,46 @@ async def whois_command(self, msg, **kwargs):
         tips_count = PlayerReport.objects.filter(to_player=player, value__gt=0).count()
 
-        # Contin
+        ##TODO: use above variables to show in !wh command
```
tutaj

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -673,11 +659,9 @@ async def add_to_queue_command(self, msg, **kwargs):
-                f'\nYou have 5 min to join the lobby.'
-            )
+
+            ##TODO - multiline arguments
```
small issue

PogSmok replied:
Nie rozumiem, co masz na myśli przez "multiline arguments".

sellmark replied:
rozbić na linie, ale ignore jak rozmawialiśmy

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -1075,78 +954,64 @@ async def admin_help_command(self, msg, **kwargs):
     async def set_name_command(self, msg, **kwargs):
         command = msg.content
         admin = kwargs['player']
+        #TODO HEREEEEEEEEEEE
```
To Twój komentarz, albo mój bardzo stary jeśli tak to usuń

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -1422,25 +1203,21 @@ async def player_join_queue(self, player, channel):
 
-            pre_str = f'\nGra #{queue.id} jest pełna!\n'
+            response += TRANSLATIONS[LANG]["proposed_balance"].format(balance_str, f' '.join(self.player_mention(p) for p in queue.players.all()), WATING_TIME_MINS)
+            ##TODO: tutaj było wysyłanie na kanał, jest gdzieś indziej teraz?
```
Tutaj

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -1524,19 +1301,12 @@ def queue_str(q: LadderQueue, show_min_mmr=True):
 
-        suffix = LadderSettings.get_solo().noob_queue_suffix
+            time_game = timeago.format(q.game_start_time, timezone.now())
+            #TODO change values in '<>' for actual values
```
To jest nierozwiązane?

sellmark reviewed (commented):
@PogSmok zostawiłem Ci komentarze w zmergowany branchu bez konfliktów

PogSmok committed:
Changed translation to include format for commands and reports

PogSmok committed:
Changed translation to include format for commands and reports

PogSmok committed:
Minor fixes

- defined WAITING_TIME_MINS
- added reports/tips to whois
- now sending response in player_join_queue()
- deleted redundant comments

sellmark committed:
Missing function

PogSmok committed:
Fixed incorrect formatting

PogSmok committed:
Return expanded roles string after updating roles

PogSmok committed:
Recent matches format fix

sellmark committed:
Getting teams from queue method

sellmark committed:
Merge branch 'feature/report-tip-commands' of github.com:sellmark/dota-discord-bot into feature/report-tip-commands

PogSmok committed:
Recent match format fix

sellmark committed:
Merge branch 'feature/report-tip-commands' of github.com:sellmark/dota-discord-bot into feature/report-tip-commands

PogSmok committed:
Include queue id in dota lobby name

PogSmok committed:
Include queue id in dota lobby name

PogSmok committed:
Finished formatting for game start

sellmark committed:
Fixes for game start and join embeds

sellmark committed:
Reports are writable on PRIV only, shown after 24 hours on webpage and discord

PogSmok committed:
Final translations version (V3)

PogSmok committed:
Remove english translation file entirely

sellmark committed:
Polishing changes for translations

sellmark committed:
Polishing changes for translations

sellmark committed:
Finest polishes

sellmark committed:
Fix bug on !reg & other latest

sellmark committed:
Typos

sellmark committed:
Bug fixes

sellmark committed:
Make polish accents and Emojis be a part of !tip, !report but also !rename. Hide Errors exposition.

sellmark committed:
Fix migration

sellmark committed:
Fix migration for good

sellmark committed:
Remove reasoning from tip/report, remove column migration

sellmark committed:
Remove duplicate welcome message(when sent from channel).
